[BUGFIX], [ENHANCEMENT] Usage Stats improvements and fixes

eugmandel opened the pull request:
### Changes proposed in this pull request:

- New usage stats event: "cli.new_ds_choice". It fires when users choose to add a new datasource in the CLI (but before the datasource is added). This will allow us to collect the data about users' intent and have an idea of success rate of adding datasources of various types.
- New usage stats event: "data_context.add_datasource". It wraps the add_datasource method in DataContext. 
- New usage stats event: "datasource.sqlalchemy.connect" - fixes the ability to get SQLAlchemy dialect of SQLAlchemy Datasources. We lost that ability when we switched to lazy loading of datasources in 0.10.9. Since that change, the dialect of SQLAlchemy Datasources could not be obtained reliably when we send "data_context.__init__" events.


### Previous Design Review notes:
-
-
-


Thank you for submitting!

eugmandel committed:
Added 3 new usage stats events:
1. cli.new_ds_choice
2. data_context.add_datasource
3. datasource.sqlalchemy.connect - fixes the ability to get SQLAlchemy dialect of SQLAlchemyDatasources

eugmandel committed:
isort

eugmandel committed:
Merge remote-tracking branch 'origin/develop' into eugene/usage_stats_events_2020_06

eugmandel committed:
changelog

jcampbell commented on great_expectations/core/usage_statistics/usage_statistics.py:
```diff
@@ -357,6 +357,23 @@ def edit_expectation_suite_usage_statistics(data_context, expectation_suite_name
+    except AttributeError:
+        data_context_id = None
+
+    datasource_anonymizer = DatasourceAnonymizer(data_context_id)
```
Can this use the handler's anonymizer to avoid instantiating a new one?

eugmandel replied:
Addressed. Thanks to @roblim!

jcampbell reviewed (approved):
Pending check on initialization of Datasource

eugmandel committed:
1. Use the handler's data source anonymizer instead of creating a new one every time.
2. usage stats messages were not sent when the wrapped method raised an exception because we obtained the reference to the handler only after the method returns.

eugmandel committed:
Merge branch 'develop' into eugene/usage_stats_events_2020_06

eugmandel committed:
Fix: get the datasource anonymizer from the handler, but if the context has no handler, create a new one

eugmandel committed:
Merge remote-tracking branch 'origin/eugene/usage_stats_events_2020_06' into eugene/usage_stats_events_2020_06

eugmandel committed:
linting

eugmandel committed:
test fix

eugmandel committed:
Merge branch 'develop' into eugene/usage_stats_events_2020_06
